cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/WorkflowStatsSLAHttpHandler.java

```diff
@@ -342,4 +348,16 @@ List<ProgramMetrics> getProgramMetricsList() {
       return programMetricsList;
     }
   }
+  /**
+   * @param namespaceId namespace Id
+   * @param appId       app Id
+   * @return latest app version
+   */
+  private String getLatestAppVersion(NamespaceId namespaceId, String appId) throws ApplicationNotFoundException {
```

Comment:
This logic should get moved into the Store to provide better abstraction. Also we can have the latest app lookup and the query of workflow information happening in the same transaction by doing so.